### PR TITLE
feat: see through Lens

### LINKED ISSUE #1053: See the feed through the lens of someone else
Sometimes it's easy to get stuck in a bubble of only the people you're following, and you might be curious about what the feed looks like for someone else. 

We are proposing adding a feature where you can choose any profile to view the feed through.

The endpoint already exists for this, just need UI/FE development.

----Design to follow today----

### PR DESCRIPTION
**Pull request by sasicodes**
## What does this PR do?

Fixes #1053 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't read the [contributing guide](https://github.com/lensterxyz/lenster/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by sasicodes** [comment]
@bigint doi

review this on side.

**Comment by bigint** [comment]
@sasicodes is it ready for review?

**Review by bigint — CHANGES_REQUESTED** [review]

File: src/components/Home/SeeThroughLens.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import getAvatar from '@lib/getAvatar';
+import { Leafwatch } from '@lib/leafwatch';
+import clsx from 'clsx';
+import type { ChangeEvent } from 'react';
+import React, { Fragment, useState } from 'react';
+import { useAppStore } from 'src/store/app';
+import { useTimelineStore } from 'src/store/timeline';
+import { SEARCH } from 'src/tracking';
+
+const SeeThroughLens = () => {
```
[the remark this answers is not in the document]
Comment (sasicodes) [reply src/components/Home/SeeThroughLens.tsx:18]: any specific reason adding it?

File: src/components/Home/Highlights.tsx [hunk 2]
```diff
@@ -12,16 +12,18 @@ import type { FC } from 'react';
...
 const Highlights: FC = () => {
   const currentProfile = useAppStore((state) => state.currentProfile);
   const txnQueue = useTransactionPersistStore((state) => state.txnQueue);
+  const seeThroughProfile = useTimelineStore((state) => state.seeThroughProfile);
 
   // Variables
-  const request = { profileId: currentProfile?.id, limit: 10 };
-  const reactionRequest = currentProfile ? { profileId: currentProfile?.id } : null;
-  const profileId = currentProfile?.id ?? null;
+  const profileId = seeThroughProfile?.id ?? currentProfile?.id ?? null;
```
Comment (harish-sethuraman) [inline src/components/Home/Highlights.tsx:24]:
```suggestion
  const profileId = seeThroughProfile?.id ?? currentProfile?.id;
```

ser? but null coersion is for adding fallback to null values?

File: src/components/Home/SeeThroughLens.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  const recommendedProfiles = recommendedProfilesToSeeThrough ?? [];
+
+  const profiles =
+    searchResults.length && searchText.length ? searchResults : recommendedProfiles.slice(0, 5);
+
+  return (
+    <Menu as="div" className="relative">
+      <Menu.Button className="rounded-md hover:bg-gray-300 p-1 hover:bg-opacity-20">
+        <span className="flex space-x-1 items-center text-sm">
+          <img
```
Comment (harish-sethuraman) [inline src/components/Home/SeeThroughLens.tsx:55]: Should we consider using next/image here? as height and width is predefined?

File: src/components/Home/SeeThroughLens.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            src={getAvatar(profile)}
+            loading="lazy"
+            className="bg-gray-200 w-5 h-5 rounded-full border dark:border-gray-700/80"
+            alt={profile?.handle}
+          />
+          <span>{seeThroughProfile ? `@${profile?.handle}` : 'My Feed'}</span>
```
Comment (harish-sethuraman) [inline src/components/Home/SeeThroughLens.tsx:61]: @bigint should we consider having a component instead of always prefixing it with `@`?  WDYT

**Comment by bigint** [comment]
We have migrated to monorepo, can you please sync with `main` branch 🙏🏼

**Pull request closed without merging**